src/org/omegat/gui/dictionaries/DictionariesTextArea.java

```diff
@@ -257,8 +259,25 @@ protected void setFoundResult(final SourceTextEntry se, final List<DictionaryEnt
 
             displayedWords.add(de.getWord().toLowerCase());
             i++;
+            if (i % textBatch == 0) {
+                pos = pos + addText(pos, txt.toString());
+                txt = new StringBuilder();
+            }
+        }
+        addText(pos, txt.toString());
+    }
+
+    private int addText(final int pos, final String txt) {
+        if (pos > 0) {
+            try {
+                getDocument().insertString(pos, txt, SimpleAttributeSet.EMPTY);
+            } catch (BadLocationException e) {
+                Log.log(e);
+            }
+        } else {
```

Comment:
Do we need this `else`? Can't the `pos == 0` case also use `insertString`?

Reply:
> Can't the pos == 0 case also use insertString?

It breaks output.

Reply:
![image](https://user-images.githubusercontent.com/123720/124387180-7792e000-dd18-11eb-9537-1f2d86eed352.png)

Reply:
It seems that JTextPane/Document need to call `setText(txt)` with significant size of html text once, then `insetText(txt)` works unexpectedly.

Reply:
Thanks. That's weird. I tried to debug why that is, but it's too much for me.